packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -435,6 +548,18 @@ class ContainerLayer extends Layer {
     assert(!child.attached);
   }
 
+  @override
+  void dropChild(AbstractNode child) {
+    markNeedsAddToScene();
+    super.dropChild(child);
+  }
+
+  @override
+  void adoptChild(AbstractNode child) {
+    markNeedsAddToScene();
+    super.adoptChild(child);
+  }
+
```

Comment:
i'd probably put this in Layer, not ContainerLayer, but I guess it doesn't matter.

Reply:
Done.